server/webapp/WEB-INF/rails/webpack/views/pages/pac/actions.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ApiResult} from "helpers/api_request_builder";
+import {MithrilViewComponent} from "jsx/mithril-component";
+import m from "mithril";
+import Stream from "mithril/stream";
+import {ConfigReposCRUD} from "models/config_repos/config_repos_crud";
+import {ConfigRepo} from "models/config_repos/types";
+import {Material} from "models/materials/types";
+import * as Buttons from "views/components/buttons";
+import css from "./styles.scss";
+
+interface Attrs {
+  material: Stream<Material>;
+  pluginId: Stream<string>;
+  loc?: LocationHandler;
+}
+
+export interface LocationHandler {
+  go: (url: string) => void;
+}
+
+class WindowLocation implements LocationHandler {
+  go(url: string) {
+    window.location.href = url;
+  }
+}
+
+export class PacActions extends MithrilViewComponent<Attrs> {
+  private location: LocationHandler = new WindowLocation();
+  private globalError: Stream<string> = Stream();
+
+  oninit(vnode: m.Vnode<Attrs, {}>) {
+    if (vnode.attrs.loc) {
+      this.location = vnode.attrs.loc;
+    }
+  }
+
+  view(vnode: m.Vnode<Attrs>): m.Children | void | null {
+    return (
+      <div class={css.actions}>
+        <Buttons.Cancel css={css} onclick={this.onCancel.bind(this)} small={false}>Cancel</Buttons.Cancel>
+
+        <div class={css.finishBtnWrapper}>
+          <div class={css.errorResponse}>{this.globalError()}</div>
+
+          <Buttons.Primary onclick={this.onSave.bind(this, vnode.attrs.pluginId(), vnode.attrs.material())} small={false}>Finish</Buttons.Primary>
+        </div>
+      </div>
+    );
+  }
+
+  onCancel(event: Event): void {
+    event.stopPropagation();
+    this.location.go("/go/pipelines");
+  }
+
+  onSave(pluginId: string, material: Material, event: Event): void {
+    event.stopPropagation();
+    event.preventDefault();
+    this.clearErrors();
+
+    const configRepo = new ConfigRepo(material.name(), pluginId, material);
```

Comment:
This is likely going to be provided by a separate user input field soon. There's a PR that will remove the "material name" and "destination" fields from the PaC material form, just FYI.

Reply:
@marques-work link to that PR so I can have a better understanding of what's gonna change?

Reply:
This is the one I'm talking about:

https://github.com/gocd/gocd/pull/6847

Reply:
@ibnc I also pushed a conflict fix to your branch

Reply:
@marques-work Okay, I see what you mean. Yeah, as I mentioned in the description of this PR, this PR is waiting on the pac id input field. I figured the material name would work fine, but it's probably better to have a separate field. Regardless, this PR won't be mergeable until that's been finished.

Reply:
@ibnc You could just add a field and label it with "Name this repository" or something, maybe?

Reply:
@ibnc If you were referring to the live validating fields, they already exist. The incantation you want to use is probably `<IdentifierInputField/>` located in `server/webapp/WEB-INF/rails/webpack/views/components/forms/common_validating_inputs.tsx`

Reply:
@marques-work That's what I'm planning on doing today. Albeit I didn't know about that field